src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,125 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.exception.RegistrationFailureException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final int MIN_VALID_AGE = 18;
+    private static final String VALID_LOGIN = "someLogin";
+    private static final String VALID_PASSWORD = "somePassword";
+    private static RegistrationService registrationService;
+    private static StorageDao storageDao;
+    private static User user;
+
+    @BeforeAll
+    static void setup() {
+        user = new User();

```

Comment:
Better to create user in before each, to make sure user is new for each test

Reply:
But we don't need a NEW user in every new test. There is @BeforeEach public void initializeValidUser() with field settings to the valid values.